### PR TITLE
Docker build

### PR DESCRIPTION
**Pull request by soerenbf**
## Purpose

Added `Dockerfile` for building and running the project using docker

### CODE REVIEW

File: frontend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+RUN yarn build
+
+# Production image, copy all the files and run next
+FROM base AS runner
+WORKDIR /app
+
+ENV NODE_ENV production
+# Uncomment the following line in case you want to disable telemetry during runtime.
+ENV NEXT_TELEMETRY_DISABLED 1
```
Comment (abizjak) [inline frontend/Dockerfile:32]: Is the comment accurate? This is setting disabled = 1, which would usually mean to set that flag.

File: frontend/README.md [hunk 2]
```diff
@@ -50,6 +50,20 @@ The project depends on a number of configuration options that can be set through
 -   `NEXT_PUBLIC_BRIDGE_MANAGER`: Hex encoded contract schema for Concordium [bridge manager contract](../concordium_contracts/bridge-manager).
 -   `NEXT_PUBLIC_CIS2_BRIDGEABLE`: Hex encoded contract schema for Concordium [cis2-bridgeable contract](../concordium_contracts/cis2-bridgeable).
 
+### Docker
+
+The project can be built and run with docker by executing
+
+```bash
+docker build -t eth-ccd-bridge-app .
```
Comment (abizjak) [inline frontend/README.md:58]: Dont' some environment variables have to be set? Either here or when running. Can we explain where they need to be set?

**Review by abizjak — APPROVED** [review]

**Comment by abizjak** [comment]
Could we add the following sample docker-compose file
```
services:
  frontend:
    image: eth-ccd-bridge-app
    build:
      context: .
      args:
        - NEXT_PUBLIC_ROOT_MANAGER_ADDRESS=0x5D531237cdCDE1f2e49dD3B290bE346cCe7d87CF
        - NEXT_PUBLIC_GENERATE_ERC20_PREDICATE_ADDRESS=0x8ae85d849167ff996c04040c44924fd364217285e4cad818292c7ac37c0a345b
    
        - NEXT_PUBLIC_BRIDGE_MANAGER_INDEX=3494
    
        - NEXT_PUBLIC_ETH_TOKEN_ADDRESS=0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
    
        - NEXT_PUBLIC_ETHEREUM_PROVIDER_NETWORK=5
        - NEXT_PUBLIC_ETHEREUM_EXPLORER_URL=https://goerli.etherscan.io/tx/{}
    
        - NEXT_PUBLIC_NETWORK_GENESIS_BLOCK_HASH=4221332d34e1694168c2a0c0b3fd0f273809612cb13d000d5c2e00e85f50f796
    
        - NEXT_PUBLIC_API_URL=http://ethbridge.testnet.concordium.com
        - NEXT_PUBLIC_CCDSCAN_URL=https://testnet.ccdscan.io/?dcount=1&dentity=transaction&dhash={}
    
        - NEXT_PUBLIC_BRIDGE_MANAGER=ffff03010000000e0000006272696467652d6d616e616765720009000000090000006772616e74526f6c650014000200000007000000616464726573731502000000070000004163636f756e7401010000000b08000000436f6e747261637401010000000c04000000726f6c6515030000000500000041646d696e02060000004d6170706572020b000000537461746553796e6365720207000000686173526f6c650214000200000007000000616464726573731502000000070000004163636f756e7401010000000b08000000436f6e747261637401010000000c04000000726f6c6515030000000500000041646d696e02060000004d6170706572020b000000537461746553796e636572021401010000000112000000726563656976655374617465557064617465001502000000070000004465706f73697401010000001400050000000200000069640504000000757365721502000000070000004163636f756e7401010000000b08000000436f6e747261637401010000000c04000000726f6f7413140000000206000000616d6f756e741b2500000008000000746f6b656e5f69641d0008000000546f6b656e4d617001010000001400030000000200000069640504000000726f6f74131400000002050000006368696c640c0a00000072656d6f7665526f6c650014000200000007000000616464726573731502000000070000004163636f756e7401010000000b08000000436f6e747261637401010000000c04000000726f6c6515030000000500000041646d696e02060000004d6170706572020b000000537461746553796e636572020900000073657450617573656404140001000000060000007061757365640115040000000e000000496e76616c6964546f6b656e49640211000000496e73756666696369656e7446756e6473020c000000556e617574686f72697a65640206000000437573746f6d0101000000150e0000000b0000005061727365506172616d7302070000004c6f6746756c6c020c0000004c6f674d616c666f726d6564020e000000436f6e74726163745061757365640213000000496e766f6b65436f6e74726163744572726f720213000000496e766f6b655472616e736665724572726f72020e000000546f6b656e4e6f744d6170706564020f000000526f6c654e6f7441737369676e656402110000005769746864726177466565546f6f4c6f7702190000004f7065726174696f6e416c726561647950726f636573736564021a0000004661696c6564557067726164654d697373696e674d6f64756c65021c0000004661696c6564557067726164654d697373696e67436f6e747261637402250000004661696c656455706772616465556e737570706f727465644d6f64756c6556657273696f6e02170000004f6e6c794163636f756e747343616e5769746864726177020c00000073657454726561737572657200140001000000070000006163636f756e740b0e00000073657457697468647261774665650014000100000006000000616d6f756e740a070000007570677261646504140002000000060000006d6f64756c651e20000000070000006d6967726174651502000000040000004e6f6e650204000000536f6d6501010000000f16011d0115040000000e000000496e76616c6964546f6b656e49640211000000496e73756666696369656e7446756e6473020c000000556e617574686f72697a65640206000000437573746f6d0101000000150e0000000b0000005061727365506172616d7302070000004c6f6746756c6c020c0000004c6f674d616c666f726d6564020e000000436f6e74726163745061757365640213000000496e766f6b65436f6e74726163744572726f720213000000496e766f6b655472616e736665724572726f72020e000000546f6b656e4e6f744d6170706564020f000000526f6c654e6f7441737369676e656402110000005769746864726177466565546f6f4c6f7702190000004f7065726174696f6e416c726561647950726f636573736564021a0000004661696c6564557067726164654d697373696e674d6f64756c65021c0000004661696c6564557067726164654d697373696e67436f6e747261637402250000004661696c656455706772616465556e737570706f727465644d6f64756c6556657273696f6e02170000004f6e6c794163636f756e747343616e576974686472617702080000007769746864726177001400040000000b0000006574685f6164647265737313140000000206000000616d6f756e741b2500000005000000746f6b656e0c08000000746f6b656e5f69641d0000
    
        - NEXT_PUBLIC_CIS2_BRIDGEABLE=ffff03010000000f000000636973322d62726964676561626c6501001400020000000300000075726c160204000000686173681502000000040000004e6f6e650204000000536f6d650101000000132000000002100000000900000062616c616e63654f6606100114000200000008000000746f6b656e5f69641d0007000000616464726573731502000000070000004163636f756e7401010000000b08000000436f6e747261637401010000000c10011b2500000015040000000e000000496e76616c6964546f6b656e49640211000000496e73756666696369656e7446756e6473020c000000556e617574686f72697a65640206000000437573746f6d0101000000150a0000000b0000005061727365506172616d7302070000004c6f6746756c6c020c0000004c6f674d616c666f726d6564020e000000436f6e74726163745061757365640213000000496e766f6b65436f6e74726163744572726f720213000000496e766f6b655472616e736665724572726f72020f000000526f6c654e6f7441737369676e6564021a0000004661696c6564557067726164654d697373696e674d6f64756c65021c0000004661696c6564557067726164654d697373696e67436f6e747261637402250000004661696c656455706772616465556e737570706f727465644d6f64756c6556657273696f6e02070000006465706f7369740414000300000007000000616464726573731502000000070000004163636f756e7401010000000b08000000436f6e747261637401010000000c06000000616d6f756e741b2500000008000000746f6b656e5f69641d0015040000000e000000496e76616c6964546f6b656e49640211000000496e73756666696369656e7446756e6473020c000000556e617574686f72697a65640206000000437573746f6d0101000000150a0000000b0000005061727365506172616d7302070000004c6f6746756c6c020c0000004c6f674d616c666f726d6564020e000000436f6e74726163745061757365640213000000496e766f6b65436f6e74726163744572726f720213000000496e766f6b655472616e736665724572726f72020f000000526f6c654e6f7441737369676e6564021a0000004661696c6564557067726164654d697373696e674d6f64756c65021c0000004661696c6564557067726164654d697373696e67436f6e747261637402250000004661696c656455706772616465556e737570706f727465644d6f64756c6556657273696f6e02090000006772616e74526f6c650414000200000007000000616464726573731502000000070000004163636f756e7401010000000b08000000436f6e747261637401010000000c04000000726f6c6515020000000500000041646d696e02070000004d616e616765720215040000000e000000496e76616c6964546f6b656e49640211000000496e73756666696369656e7446756e6473020c000000556e617574686f72697a65640206000000437573746f6d0101000000150a0000000b0000005061727365506172616d7302070000004c6f6746756c6c020c0000004c6f674d616c666f726d6564020e000000436f6e74726163745061757365640213000000496e766f6b65436f6e74726163744572726f720213000000496e766f6b655472616e736665724572726f72020f000000526f6c654e6f7441737369676e6564021a0000004661696c6564557067726164654d697373696e674d6f64756c65021c0000004661696c6564557067726164654d697373696e67436f6e747261637402250000004661696c656455706772616465556e737570706f727465644d6f64756c6556657273696f6e0207000000686173526f6c650614000200000007000000616464726573731502000000070000004163636f756e7401010000000b08000000436f6e747261637401010000000c04000000726f6c6515020000000500000041646d696e02070000004d616e61676572021401010000000115040000000e000000496e76616c6964546f6b656e49640211000000496e73756666696369656e7446756e6473020c000000556e617574686f72697a65640206000000437573746f6d0101000000150a0000000b0000005061727365506172616d7302070000004c6f6746756c6c020c0000004c6f674d616c666f726d6564020e000000436f6e74726163745061757365640213000000496e766f6b65436f6e74726163744572726f720213000000496e766f6b655472616e736665724572726f72020f000000526f6c654e6f7441737369676e6564021a0000004661696c6564557067726164654d697373696e674d6f64756c65021c0000004661696c6564557067726164654d697373696e67436f6e747261637402250000004661696c656455706772616465556e737570706f727465644d6f64756c6556657273696f6e020a0000006f70657261746f724f66061001140002000000050000006f776e65721502000000070000004163636f756e7401010000000b08000000436f6e747261637401010000000c07000000616464726573731502000000070000004163636f756e7401010000000b08000000436f6e747261637401010000000c10010115040000000e000000496e76616c6964546f6b656e49640211000000496e73756666696369656e7446756e6473020c000000556e617574686f72697a65640206000000437573746f6d0101000000150a0000000b0000005061727365506172616d7302070000004c6f6746756c6c020c0000004c6f674d616c666f726d6564020e000000436f6e74726163745061757365640213000000496e766f6b65436f6e74726163744572726f720213000000496e766f6b655472616e736665724572726f72020f000000526f6c654e6f7441737369676e6564021a0000004661696c6564557067726164654d697373696e674d6f64756c65021c0000004661696c6564557067726164654d697373696e67436f6e747261637402250000004661696c656455706772616465556e737570706f727465644d6f64756c6556657273696f6e020a00000072656d6f7665526f6c650414000200000007000000616464726573731502000000070000004163636f756e7401010000000b08000000436f6e747261637401010000000c04000000726f6c6515020000000500000041646d696e02070000004d616e616765720215040000000e000000496e76616c6964546f6b656e49640211000000496e73756666696369656e7446756e6473020c000000556e617574686f72697a65640206000000437573746f6d0101000000150a0000000b0000005061727365506172616d7302070000004c6f6746756c6c020c0000004c6f674d616c666f726d6564020e000000436f6e74726163745061757365640213000000496e766f6b65436f6e74726163744572726f720213000000496e766f6b655472616e736665724572726f72020f000000526f6c654e6f7441737369676e6564021a0000004661696c6564557067726164654d697373696e674d6f64756c65021c0000004661696c6564557067726164654d697373696e67436f6e747261637402250000004661696c656455706772616465556e737570706f727465644d6f64756c6556657273696f6e020f000000736574496d706c656d656e746f72730414000200000002000000696416000c000000696d706c656d656e746f727310020c15040000000e000000496e76616c6964546f6b656e49640211000000496e73756666696369656e7446756e6473020c000000556e617574686f72697a65640206000000437573746f6d0101000000150a0000000b0000005061727365506172616d7302070000004c6f6746756c6c020c0000004c6f674d616c666f726d6564020e000000436f6e74726163745061757365640213000000496e766f6b65436f6e74726163744572726f720213000000496e766f6b655472616e736665724572726f72020f000000526f6c654e6f7441737369676e6564021a0000004661696c6564557067726164654d697373696e674d6f64756c65021c0000004661696c6564557067726164654d697373696e67436f6e747261637402250000004661696c656455706772616465556e737570706f727465644d6f64756c6556657273696f6e020900000073657450617573656404140001000000060000007061757365640115040000000e000000496e76616c6964546f6b656e49640211000000496e73756666696369656e7446756e6473020c000000556e617574686f72697a65640206000000437573746f6d0101000000150a0000000b0000005061727365506172616d7302070000004c6f6746756c6c020c0000004c6f674d616c666f726d6564020e000000436f6e74726163745061757365640213000000496e766f6b65436f6e74726163744572726f720213000000496e766f6b655472616e736665724572726f72020f000000526f6c654e6f7441737369676e6564021a0000004661696c6564557067726164654d697373696e674d6f64756c65021c0000004661696c6564557067726164654d697373696e67436f6e747261637402250000004661696c656455706772616465556e737570706f727465644d6f64756c6556657273696f6e0213000000736574546f6b656e4d6574616461746155726c041400020000000300000075726c160204000000686173681502000000040000004e6f6e650204000000536f6d65010100000013200000000215040000000e000000496e76616c6964546f6b656e49640211000000496e73756666696369656e7446756e6473020c000000556e617574686f72697a65640206000000437573746f6d0101000000150a0000000b0000005061727365506172616d7302070000004c6f6746756c6c020c0000004c6f674d616c666f726d6564020e000000436f6e74726163745061757365640213000000496e766f6b65436f6e74726163744572726f720213000000496e766f6b655472616e736665724572726f72020f000000526f6c654e6f7441737369676e6564021a0000004661696c6564557067726164654d697373696e674d6f64756c65021c0000004661696c6564557067726164654d697373696e67436f6e747261637402250000004661696c656455706772616465556e737570706f727465644d6f64756c6556657273696f6e0208000000737570706f727473061001160010011503000000090000004e6f537570706f72740207000000537570706f72740209000000537570706f72744279010100000010000c15040000000e000000496e76616c6964546f6b656e49640211000000496e73756666696369656e7446756e6473020c000000556e617574686f72697a65640206000000437573746f6d0101000000150a0000000b0000005061727365506172616d7302070000004c6f6746756c6c020c0000004c6f674d616c666f726d6564020e000000436f6e74726163745061757365640213000000496e766f6b65436f6e74726163744572726f720213000000496e766f6b655472616e736665724572726f72020f000000526f6c654e6f7441737369676e6564021a0000004661696c6564557067726164654d697373696e674d6f64756c65021c0000004661696c6564557067726164654d697373696e67436f6e747261637402250000004661696c656455706772616465556e737570706f727465644d6f64756c6556657273696f6e020d000000746f6b656e4d657461646174610610011d0010011400020000000300000075726c160104000000686173681502000000040000004e6f6e650204000000536f6d65010100000013200000000215040000000e000000496e76616c6964546f6b656e49640211000000496e73756666696369656e7446756e6473020c000000556e617574686f72697a65640206000000437573746f6d0101000000150a0000000b0000005061727365506172616d7302070000004c6f6746756c6c020c0000004c6f674d616c666f726d6564020e000000436f6e74726163745061757365640213000000496e766f6b65436f6e74726163744572726f720213000000496e766f6b655472616e736665724572726f72020f000000526f6c654e6f7441737369676e6564021a0000004661696c6564557067726164654d697373696e674d6f64756c65021c0000004661696c6564557067726164654d697373696e67436f6e747261637402250000004661696c656455706772616465556e737570706f727465644d6f64756c6556657273696f6e02080000007472616e7366657204100114000500000008000000746f6b656e5f69641d0006000000616d6f756e741b250000000400000066726f6d1502000000070000004163636f756e7401010000000b08000000436f6e747261637401010000000c02000000746f1502000000070000004163636f756e7401010000000b08000000436f6e747261637401020000000c160104000000646174611d0115040000000e000000496e76616c6964546f6b656e49640211000000496e73756666696369656e7446756e6473020c000000556e617574686f72697a65640206000000437573746f6d0101000000150a0000000b0000005061727365506172616d7302070000004c6f6746756c6c020c0000004c6f674d616c666f726d6564020e000000436f6e74726163745061757365640213000000496e766f6b65436f6e74726163744572726f720213000000496e766f6b655472616e736665724572726f72020f000000526f6c654e6f7441737369676e6564021a0000004661696c6564557067726164654d697373696e674d6f64756c65021c0000004661696c6564557067726164654d697373696e67436f6e747261637402250000004661696c656455706772616465556e737570706f727465644d6f64756c6556657273696f6e020e0000007570646174654f70657261746f720410011400020000000600000075706461746515020000000600000052656d6f7665020300000041646402080000006f70657261746f721502000000070000004163636f756e7401010000000b08000000436f6e747261637401010000000c15040000000e000000496e76616c6964546f6b656e49640211000000496e73756666696369656e7446756e6473020c000000556e617574686f72697a65640206000000437573746f6d0101000000150a0000000b0000005061727365506172616d7302070000004c6f6746756c6c020c0000004c6f674d616c666f726d6564020e000000436f6e74726163745061757365640213000000496e766f6b65436f6e74726163744572726f720213000000496e766f6b655472616e736665724572726f72020f000000526f6c654e6f7441737369676e6564021a0000004661696c6564557067726164654d697373696e674d6f64756c65021c0000004661696c6564557067726164654d697373696e67436f6e747261637402250000004661696c656455706772616465556e737570706f727465644d6f64756c6556657273696f6e02070000007570677261646504140002000000060000006d6f64756c651e20000000070000006d6967726174651502000000040000004e6f6e650204000000536f6d6501010000000f16011d0115040000000e000000496e76616c6964546f6b656e49640211000000496e73756666696369656e7446756e6473020c000000556e617574686f72697a65640206000000437573746f6d0101000000150a0000000b0000005061727365506172616d7302070000004c6f6746756c6c020c0000004c6f674d616c666f726d6564020e000000436f6e74726163745061757365640213000000496e766f6b65436f6e74726163744572726f720213000000496e766f6b655472616e736665724572726f72020f000000526f6c654e6f7441737369676e6564021a0000004661696c6564557067726164654d697373696e674d6f64756c65021c0000004661696c6564557067726164654d697373696e67436f6e747261637402250000004661696c656455706772616465556e737570706f727465644d6f64756c6556657273696f6e020400000076696577051400020000000c0000006d657461646174615f75726c1400020000000300000075726c160104000000686173681502000000040000004e6f6e650204000000536f6d650101000000132000000002060000007061757365640115040000000e000000496e76616c6964546f6b656e49640211000000496e73756666696369656e7446756e6473020c000000556e617574686f72697a65640206000000437573746f6d0101000000150a0000000b0000005061727365506172616d7302070000004c6f6746756c6c020c0000004c6f674d616c666f726d6564020e000000436f6e74726163745061757365640213000000496e766f6b65436f6e74726163744572726f720213000000496e766f6b655472616e736665724572726f72020f000000526f6c654e6f7441737369676e6564021a0000004661696c6564557067726164654d697373696e674d6f64756c65021c0000004661696c6564557067726164654d697373696e67436f6e747261637402250000004661696c656455706772616465556e737570706f727465644d6f64756c6556657273696f6e020800000077697468647261770414000300000007000000616464726573731502000000070000004163636f756e7401010000000b08000000436f6e747261637401010000000c06000000616d6f756e741b2500000008000000746f6b656e5f69641d0015040000000e000000496e76616c6964546f6b656e49640211000000496e73756666696369656e7446756e6473020c000000556e617574686f72697a65640206000000437573746f6d0101000000150a0000000b0000005061727365506172616d7302070000004c6f6746756c6c020c0000004c6f674d616c666f726d6564020e000000436f6e74726163745061757365640213000000496e766f6b65436f6e74726163744572726f720213000000496e766f6b655472616e736665724572726f72020f000000526f6c654e6f7441737369676e6564021a0000004661696c6564557067726164654d697373696e674d6f64756c65021c0000004661696c6564557067726164654d697373696e67436f6e747261637402250000004661696c656455706772616465556e737570706f727465644d6f64756c6556657273696f6e0200
    
```
and explain that to build run
```
docker-compose build
```

The environment variables msut be set during build only.

File: frontend/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
```
Comment (abizjak) [inline frontend/Dockerfile:33]:
```suggestion

ARG NEXT_PUBLIC_ROOT_MANAGER_ADDRESS

ARG NEXT_PUBLIC_GENERATE_ERC20_PREDICATE_ADDRESS

ARG NEXT_PUBLIC_BRIDGE_MANAGER_INDEX

ARG NEXT_PUBLIC_ETH_TOKEN_ADDRESS

ARG NEXT_PUBLIC_ETHEREUM_PROVIDER_NETWORK

ARG NEXT_PUBLIC_ETHEREUM_EXPLORER_URL

ARG NEXT_PUBLIC_NETWORK_GENESIS_BLOCK_HASH

ARG NEXT_PUBLIC_API_URL

ARG NEXT_PUBLIC_CCDSCAN_URL

ARG NEXT_PUBLIC_BRIDGE_MANAGER

ARG NEXT_PUBLIC_CIS2_BRIDGEABLE

```

File: frontend/Dockerfile (hunk 1 above)
Comment (soerenbf) [reply frontend/Dockerfile:32]: Yes, the comment was there before the line was uncommented. I'll change it to explain what happens instead

File: frontend/README.md (hunk 2 above)
Comment (soerenbf) [reply frontend/README.md:58]: Yes, good point.

**Pull request merged**